src/api/app/controllers/source_controller.rb

```diff
@@ -336,14 +336,14 @@ def require_package
     #--------------------
     @deleted_package = params.key?(:deleted)
 
+    @target_package_name = params[:package]
+
     # FIXME: for OBS 3, api of branch and copy calls have target and source in the opposite place
-    if params[:cmd].in?(%w[branch release])
-      @target_package_name = params[:package]
+    if params[:cmd].in?(%w[branch fork release])
       @target_project_name = params[:target_project] # might be nil
       @target_package_name = params[:target_package] if params[:target_package]
```

Comment:
Adding the `fork` command to this `if` condition will make it possible to read the `target_package` query parameter and overwrite the `package` parameter from the path.

If this is what we really want, the `target_package` query parameter should added to the `...?cmd=fork` endpoint API documentation.

Reply:
pushed